src/btrfsbalancer.h

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Martin Grimme <martin.grimme@gmail.com>
+** All rights reserved.
+**
+** This file is part of the btrfs-balancer package.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+
+#ifndef BTRFSBALANCER_H
+#define BTRFSBALANCER_H
+
+#include "btrfs.h"
+
+#include <QObject>
+#include <QString>
+
+class BtrfsBalancer : public QObject
+{
+    Q_OBJECT
+public:
+    enum Status
+    {
+        READY = 0,
+        BALANCING
+    };
+
+    explicit BtrfsBalancer(QObject *parent = 0);
+
+    /* Checks the status of the balancer.
+     * Emits the status signal.
+     */
+    void checkStatus();
+
+    /* Checks the filesystem allocation.
+     * Emits the allocation signal.
+     */
+    void checkAllocation();
+
+    /* Starts the balancing operation.
+     * Emits finished signal after success or failure.
+     * Emits progress signals inbetween.
+     */
+    void balance();
```

Comment:
Again here, I'd name it `startBalance()` so it's obvious that it just starts the balance operation, and the function returning doesn't mean that the balance has been finished. (also more consistent with `verbNoun`-style of `checkStatus` and `checkAllocation`)